org.lflang/src/org/lflang/FileConfig.java

```diff
@@ -434,34 +435,132 @@ public static void copyDirectory(Path src, Path dest) throws IOException {
+     *  @param destination The file system path that the source file is copied to.
+     * @throws IOException 
+     */
+    public void copyFileFromClassPath(String source, String destination) throws IOException {
+        InputStream sourceStream = this.getClass().getResourceAsStream(source);
+
+        if (sourceStream == null) {
+            throw new IOException(
+                "A required target resource could not be found: " + source + "\n" +
+                    "Perhaps a git submodule is missing or not up to date.\n" +
+                    "See https://github.com/icyphy/lingua-franca/wiki/downloading-and-building#clone-the-lingua-franca-repository.\n" +
+                    "Also try to refresh and clean the project explorer if working from eclipse.");
+        }
+
+        // Copy the file.
+        try {
+            // Make sure the directory exists
+            File destFile = new File(destination);
+            destFile.getParentFile().mkdirs();
+
+            Files.copy(sourceStream, Paths.get(destination), StandardCopyOption.REPLACE_EXISTING);
+        } catch (IOException ex) {
+            throw new IOException(
+                "A required target resource could not be copied: " + source + "\n" +
+                    "Perhaps a git submodule is missing or not up to date.\n" +
+                    "See https://github.com/icyphy/lingua-franca/wiki/downloading-and-building#clone-the-lingua-franca-repository.",
+                ex);
+        } finally {
+            sourceStream.close();
+        }
+    }
+
+    /**
+     * Copy a list of files from a given source directory to a given destination directory.
+     * @param srcDir The directory to copy files from.
+     * @param dstDir The directory to copy files to.
+     * @param files The list of files to copy.
+     * @throws IOException 
+     */
+    public void copyFilesFromClassPath(String srcDir, String dstDir, List<String> files) throws IOException {
+        for (String file : files) {
+            copyFileFromClassPath(srcDir + '/' + file, dstDir + File.separator + file);
+        }
+    }
+    
+   /**
+    * Copy the 'fileName' from the 'srcDirectory' to the 'destinationDirectory'.
+    * This function has a fallback search mechanism, where if `fileName` is not
+    * found in the `srcDirectory`, it will try to find `fileName` via the following procedure:
+    * 1- Search in LF_CLASSPATH. @see findFile()
+    * 2- Search in CLASSPATH. @see findFile()
+    * 3- Search for 'fileName' as a resource.
+    *  That means the `fileName` can be '/path/to/class/resource'. @see java.lang.Class.getResourceAsStream()
+    * 
+    * @param fileName Name of the file
+    * @param srcDir Where the file is currently located
+    * @param dstDir Where the file should be placed
+    * @return The name of the file in destinationDirectory
+    */
+   public String copyFileOrResource(String fileName, Path srcDir, Path dstDir) {
```

Comment:
Why the three arguments and not just two as in `copyFile()` and `copyFileFromClasspath()`?

Reply:
And also this method should probably be static

Reply:
> Why the three arguments and not just two as in `copyFile()` and `copyFileFromClasspath()`?

Just because of `findFile` since it takes in the `fileName` and the `srcDir` separately. We can attempt to extract those from a source path like the functions that you have mentioned but that can be error-prone. This keeps the logic of this function simple.

> And also this method should probably be static

This can't be static because it uses `copyFileFromClassPath()` which needs a class object to exist I think.

Reply:
If the source path is a `Path` object, we can simply use `getFileName()` to get the name of the file and `getParent()` to get its directory. Or is there something more complicated that I am missing here?

Reply:
Ah, okay. But I think this signature is still more convenient for my use case.
The `files` and `cmake-include` target properties take a file name, and the `srcDir` is the src directory of the relevant `.lf` file. If we switch to a source path, then we need to join the file name and `srcDir` before calling this function.

Should we prioritize the signature?